h2/src/main/org/h2/store/fs/disk/FilePathDisk.java

```diff
@@ -55,23 +61,35 @@ public FilePathDisk getPath(String path) {
         return p;
     }
 
+
     @Override
     public long size() {
         if (name.startsWith(CLASSPATH_PREFIX)) {
+            String path = this.name.substring("classpath:".length());
+            if (!path.startsWith("/")) {
+                path = "/" + path;
+            }
+            URL url = this.getClass().getResource(path);
             try {
-                String fileName = name.substring(CLASSPATH_PREFIX.length());
-                // Force absolute resolution in Class.getResource
-                if (!fileName.startsWith("/")) {
-                    fileName = "/" + fileName;
+                URI uri = url.toURI();
```

Comment:
There was a null check for result of `getResource()`, it should be restored. If it is null, 0L can be returned immediately.

Reply:
I think I wrote a misleading comment during the first review. I didn't check how exactly this situation was handled.

( `catch` block will return 0L anyway, but usually it is better to avoid exceptions where possible.)

Reply:
Added null check.